### PR TITLE
ARROW-13912: [R] TrimOptions implementation breaks test-r-minimal-build due to dependencies

### PR DESCRIPTION
**Pull request by nealrichardson**
Also skips a puzzling old R version test failure from ARROW-13740

cc @thisisnic

### CODE REVIEW

**Comment by nealrichardson** [comment]
@github-actions crossbow submit test-r-minimal-build test-r-versions

**Comment by nealrichardson** [comment]
@github-actions crossbow submit test-r-versions

**Comment by nealrichardson** [comment]
@github-actions crossbow submit test-r-versions

File: r/tests/testthat/test-compute-no-bindings.R [hunk 1]
```diff
@@ -15,6 +15,8 @@
 # specific language governing permissions and limitations
 # under the License.
 
+skip_if_not_available("utf8proc")
+
 test_that("non-bound compute kernels using TrimOptions", {
```
Comment (thisisnic) [inline r/tests/testthat/test-compute-no-bindings.R:20]:
```suggestion

test_that("non-bound compute kernels using TrimOptions", {

skip_if_not_available("utf8proc")
```

**Pull request closed without merging**